zebra-network/src/peer/handshake.rs

```diff
@@ -596,12 +596,16 @@ where
     // timeout, and the async mutex will be released when the task times
     // out.
     //
-    // Duplicate nonces don't matter here, because 64-bit random collisions are very rare.
-    // If they happen, we're probably replacing a leftover nonce from a failed connection,
-    // which wasn't cleaned up when it closed.
+    // START CRITICAL SECTION
```

Comment:
Can you explain why this is considered critical section and if the plan is to use this "tag" for other critical zones of our codebase ?

Reply:
https://en.wikipedia.org/wiki/Critical_section

Usually we just use a block with a comment above it, I can switch to that if you'd like.